src/asn1.ml

```diff
@@ -528,6 +596,17 @@ struct
     map f g @@ sequence2
       (required ~label:"algorithm" Algo.grammar)
       (required ~label:"parameters" EC.Params.grammar)
+
+  let dh_grammar =
+    let open Asn in
+    let f = function
+      | Algo.DH, params -> params
+      | Algo.DH, _ -> parse_error "Algorithm OID valid, but something else went wrong."
```

Comment:
I'd say drop this line since this error message won't make sense when using the generic `decode`.